rebasebot/bot.py

```diff
@@ -149,11 +149,22 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
                             "--ancestry-path", f"{merge_base}..dest/{dest.branch}")
     logging.info("Picking commits: \n%s", commits)
 
+    go_mod_commits = []
+
     for commit in commits.splitlines():
         # Commit contains the message for logging purposes,
         # trim on the first space to get just the commit sha
         sha, commit_message = commit.split(" - ")
 
+        if update_go_modules:
+            # If we find a commit with such name, we know that it is a go mod update commit
+            # We then replace <carry> flag for <drop> flag
+            # and append such commit to a list of commits that we want to prune
+            if commit_message.find("UPSTREAM: <carry>: Updating and vendoring go modules "
```

Comment:
I don't think this check is working correctly, I did some manual testing and this clause identified all commits. How about just changing to a bard `commit_message == ...`?

Reply:
Changed it as you mentioned. Makes sense to me.